pandas/tests/test_groupby.py

```diff
@@ -4594,9 +4594,6 @@ def test_frame_groupby_plot_boxplot(self):
         _skip_if_mpl_not_installed()
 
         import matplotlib.pyplot as plt
-        import matplotlib as mpl
```

Comment:
The reason it causes a warning is because it is called after `import matplotlib.pyplot as plt`. 
I was looking at the test_graphics, and there they also set the matplotlib backend, but in the test to skip it (https://github.com/sinhrks/pandas/blob/plot_doc/pandas/util/testing.py#L180). So you can also put it in the `_skip_if_mpl_not_installed` here. Only you have to import matplotlib in that test, and not matplotlib.pyplot